skfda/representation/basis.py

```diff
@@ -1589,38 +1592,33 @@ def __init__(self, basis, coefficients, *, dataset_label=None,
         super().__init__(extrapolation, dataset_label, axes_labels, keepdims)
 
     @classmethod
-    def from_data(cls, data_matrix, sample_points, basis, weight_matrix=None,
-                  smoothness_parameter=0, penalty_degree=None,
-                  penalty_coefficients=None, penalty_matrix=None,
+    def from_data(cls, data_matrix, sample_points, basis,
```

Comment:
Why not keep all possible parameters in the ``BasisSmoother`` class?

Reply:
Because it is not the same, conceptually, to change the representation than to smooth the data. It was also Alberto's opinion.

Reply:
Okay then. Although the conversion to bases already performs a smoothing. I would consider allowing arguments to pass through ``**kwargs``.

Reply:
I know that there is an implicit smoothing, but I also agree with Alberto in that, if you want to add a penalization, it is more clean to use the smoother.